wandb/apis/public/artifacts.py

```diff
@@ -501,8 +523,28 @@ def delete(self):
             )
         self.client.execute(mutation, variable_values={"id": self.id})
 
+    @property
+    def description(self):
+        """A description of the artifact collection."""
+        return self._description
+
+    @property
+    def tags(self):
+        """The tags associated with the artifact collection."""
+        return self._tags
+
+    @property
+    def name(self):
+        """The name of the artifact collection."""
+        return self._name
+
+    @property
+    def type(self):
+        """The type of the artifact collection."""
+        return self._type
+
     def __repr__(self):
-        return f"<ArtifactCollection {self.name} ({self.type})>"
+        return f"<ArtifactCollection {self._saved_name} ({self._saved_type})>"
```

Comment:
I think there's an argument for both ways, but I feel like it's more intuitive here to use `self.type` and return the new type, that way `ac.type = newtype; print(ac)` reports the new type rather than the old value.